jonatack,2021-07-19 15:10:10,Approach ACK,https://github.com/bitcoin/bitcoin/pull/22500#issuecomment-882628295,882628295,
DrahtBot,2021-07-19 18:49:04,"<!--e57a25ab6845829454e8d69fc972939a-->\n\nThe following sections might be updated with supplementary metadata relevant to reviewers and maintainers.\n\n<!--174a7506f384e20aa4161008e828411d-->\n### Conflicts\nReviewers, this pull request conflicts with the following ones:\n\n* #22498 by MarcoFalke\n* #22479 by jonatack\n* #21940 by MarcoFalke\n* #20233 by jnewbery\n\nIf you consider this pull requ",https://github.com/bitcoin/bitcoin/pull/22500#issuecomment-882777090,882777090,
jonatack,2021-07-20 10:10:59,"re-ACK fa4bfef4798466e986569baf0d8e3213f17cafc8 have been testing this rebased to current master with DEBUG_ADDRMAN defined\n\ncould squash\n",https://github.com/bitcoin/bitcoin/pull/22500#issuecomment-883272520,883272520,
jnewbery,2021-07-21 09:32:21,"~I've added these two commits to #20233, since the two PRs are doing overlapping things.~\n\nEDIT: removed them again after discussing with Marco.",https://github.com/bitcoin/bitcoin/pull/22500#issuecomment-884041918,884041918,
lsilva01,2021-07-21 20:31:42,"If `DEBUG_ADDRMAN` is set, the following error will occur when `addrman_tests` is run:\n```\n$ src/test/test_bitcoin --run_test=addrman_tests --log_level=unit_scope\nRunning 18 test cases...\nEntering test module ""Bitcoin Core Test Suite""\ntest/addrman_tests.cpp(124): Entering test suite ""addrman_tests""\ntest/addrman_tests.cpp(126): Entering test case ""addrman_simple""\ntest_bitcoin: addrman",https://github.com/bitcoin/bitcoin/pull/22500#issuecomment-884479640,884479640,
jnewbery,2021-07-21 20:38:06,"> If DEBUG_ADDRMAN is set, the following error will occur when addrman_tests is run ...\n\n@lsilva01 this is fixed in the first commit of #20233.",https://github.com/bitcoin/bitcoin/pull/22500#issuecomment-884483441,884483441,
practicalswift,2021-07-24 19:39:36,Concept ACK,https://github.com/bitcoin/bitcoin/pull/22500#issuecomment-886102463,886102463,
vasild,2021-07-26 08:48:17,"This will make it inconvenient to do something other than `abort()` on `Check()` failure.\n\nFor example, it might make sense to do the check right after `Unserialize()` and if it fails - throw `std::ios_base::failure` as if disk corruption has occurred.\n\nIn general `assert()` should be used when a possible programming error has been detected, not on bad input.",https://github.com/bitcoin/bitcoin/pull/22500#issuecomment-886506426,886506426,
jonatack,2021-07-19 16:13:20,"nit, spacing here is off by one\n",https://github.com/bitcoin/bitcoin/pull/22500#discussion_r672440473,672440473,src/addrman.cpp
jonatack,2021-07-19 16:13:45,"```diff\n     //! Consistency check\n-    void Check()\n-        EXCLUSIVE_LOCKS_REQUIRED(cs);\n+    void Check() EXCLUSIVE_LOCKS_REQUIRED(cs);\n```\n",https://github.com/bitcoin/bitcoin/pull/22500#discussion_r672440785,672440785,src/addrman.h
